tests/test_estimated_degree.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import absolute_import, print_function, division
+import contextlib
+import logging
+
+import pytest
+
+import ufl
+from tsfc import compile_form
+from tsfc.logging import logger
+
+
+class MockHandler(logging.Handler):
+    def emit(self, record):
+        raise RuntimeError()
+
+
+@contextlib.contextmanager
+def push_handler(logger, handler):
+    logger.addHandler(handler)
+    yield
+    logger.removeHandler(handler)
```

Comment:
What is the point of a context manager if I manually have to write out try-finally clauses? It's just stupid. They could have, and should have done it in `contextlib.contextmanager`.